src/remote_control.cpp

```diff
@@ -148,8 +141,10 @@ bool RemoteControl::waitForNextFullStep(const std::string& caption)
 
 bool RemoteControl::waitForNextStepCommon(const std::string& caption, bool autonomous)
 {
+  std::unique_lock<std::mutex> wait_lock(mutex_);
```

Comment:
Why the mix of `unique_lock` and `lock_guard` in `setReadyForNextStep()`? Since `mutex_` is a simple `std::mutex`, I'd use `std::scoped_lock` for both.

Reply:
`wait_next_step_` is an `std::condition_variable` whose `wait` function requires `unique_lock`, also I think my default compiler doesn't seem to be happy with scoped_lock in C++17 and needs? additional configuration.

Reply:
`scoped_lock` is accessible with C++17 and this package is still on C++14. You're right about the condition variable, though.